src/common/Kokkos_ArithTraits.hpp

```diff
@@ -1091,104 +1094,74 @@ class ArithTraits<float> {
   static const bool is_complex     = false;
 
   static constexpr bool has_infinity = true;
-  static KOKKOS_FORCEINLINE_FUNCTION float infinity() { return HUGE_VALF; }
+  static KOKKOS_FORCEINLINE_FUNCTION float infinity() {
+    return Kokkos::Experimental::infinity<float>::value;
+  }
 
   static KOKKOS_FORCEINLINE_FUNCTION bool isInf(const float x) {
-#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
-    using std::isinf;
-#elif defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_SYCL)
-    using sycl::isinf;
-#endif
-    return isinf(x);
+    return Kokkos::isinf(x);
   }
   static KOKKOS_FORCEINLINE_FUNCTION bool isNan(const float x) {
-#ifdef KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST
-    using std::isnan;
-#elif defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_SYCL)
-    using sycl::isnan;
-#endif
-    return isnan(x);
+    return Kokkos::isnan(x);
   }
   static KOKKOS_FORCEINLINE_FUNCTION mag_type abs(const float x) {
-    return ::fabs(x);
+    return Kokkos::abs(x);
   }
-  static KOKKOS_FORCEINLINE_FUNCTION float zero() { return 0.0; }
-  static KOKKOS_FORCEINLINE_FUNCTION float one() { return 1.0; }
-  static KOKKOS_FORCEINLINE_FUNCTION float min() { return -FLT_MAX; }
-  static KOKKOS_FORCEINLINE_FUNCTION float max() { return FLT_MAX; }
+  static KOKKOS_FORCEINLINE_FUNCTION float zero() { return Kokkos::reduction_identity<float>::sum(); }
+  static KOKKOS_FORCEINLINE_FUNCTION float one() { return Kokkos::reduction_identity<float>::prod(); }
```

Comment:
Any good reason why that is not
```suggestion
  static KOKKOS_FORCEINLINE_FUNCTION float zero() { return 0; }
  static KOKKOS_FORCEINLINE_FUNCTION float one() { return 1; }
```

Reply:
Not really, it was visually helpful to use the Kokkos function to check that all the calls are using the new implementation instead of the old one. Also if there is ever a problem I get to blame Kokkos Core : )

Reply:
But `reduction_identity` is not part of the numeric traits

Reply:
I meant not part of the facility that was added to replace `std::numeric_limits` and `Kokkos::ArithmeticTraits`.

Reply:
I updated this to now have all floating point types re-factored.
I just went with `static_cast<some_type>(1.0)` instead of the `reduction_identity` which I think is reasonable.